components/brave_ads/core/internal/ads/serving/choose/sample_ads.h

```diff
@@ -16,16 +17,28 @@
 
 namespace ads {
 
+template <typename T>
+T CalculateNormalizingConstantBase(const std::vector<T> scores) {
+  T normalizing_constant = 0.0;
+  for (const auto& votes : scores) {
+    normalizing_constant += votes;
+  }
+
+  return normalizing_constant;
+}
+
 template <typename T>
 double CalculateNormalizingConstant(
     const CreativeAdPredictorMap<T>& creative_ad_predictors) {
-  double normalizing_constant = 0.0;
+  std::vector<double> scores;
+  scores.assign(creative_ad_predictors.size(), 0);
```

Comment:
std container is already initialized with 0's. Can we please change to `std::vector<double> scores(creative_ad_predictors.size()`